### ISSUE TITLE
Caught KeyError in DataLoader worker process 0.

### ISSUE THREAD

**Issue by yang-Michael** [daveredrum/Scan2Cap#3]
Hello, I am following the steps in "Setup" to try to run the code, but I encountered the following problem, can you help me solve it?
My machine environment is: Ubuntu18.04, Michine Memory 8G, GPU 12G(Tesla K80), Cuda 10.2, Python 3.7.0, PyTorch 1.8.0, etc.

`python ./scripts/train.py --debug --use_color --use_normal --use_topdown --use_relation --use_orientation --num_graph_steps 2 --num_locals 10 --batch_size 12 --epoch 50`

I typed the above command on the command line just for debugging and got the following error:
![image](https://user-images.githubusercontent.com/29976881/126902615-bdf4ab35-76c9-453a-a38a-04f06d505437.png)

In addition, I find that my machine memory is a little small, which is one of the reasons why the program can't run. But, confusingly, in debug mode, where only one piece of data is loaded, the above problems still occur. As shown in the figure below, I did not download the data in Step 5. Is that the cause of the problem?
**Notice that I replaced "--use_multiview" with "--use_color" in the command**

![image](https://user-images.githubusercontent.com/29976881/126903112-06396eb1-fd2e-4d6c-8f58-28ba8c0fae2c.png)

I am looking forward to your early reply. Thank you very much

**Comment by daveredrum** [comment]
Hi @yang-Michael, I noticed that you added the flag `--debug` - Can you remove it and try again? I believe there must be some issues with the debug mode.

**Comment by yang-Michael** [comment]
Hello, Dr. Chen, thank you for your reply. I will give it a try. But I have one more problem: if I hadn't used the data in step 5 ((Optional) Pre-process the multiview features from ENet.), I would have executed the following command:


python ./scripts/train.py --use_color --use_normal --use_topdown --use_relation --use_orientation --num_graph_steps 2 --num_locals 10 --batch_size 12 --epoch 50


Notice that I replaced "--use_multiview" with "--use_color" in the command


Is that right?&nbsp;Is this part of the data necessary for training?

Maybe these are silly questions, but to be honest, I don't have enough storage space on my computer, so I wanted to know the answer straight away.


Thank you again.

Minyang Xu
WUHAN UNIVERSITY&nbsp;&nbsp;
***@***.***
Number:18571459519


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "daveredrum/Scan2Cap"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月26日(星期一) 下午5:56
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [daveredrum/Scan2Cap] Caught KeyError in DataLoader worker process 0. (#3)





 
Hi @yang-Michael, I noticed that you added the flag --debug - Can you remove it and try again? I believe there must be some issues with the debug mode.

**Comment by daveredrum** [comment]
Replacing the `--use_multiview` flag with `--use_color` is absolutely valid - Note you might want to add `--use_normal` to your current command, as using raw RGB values plus point normals are shown to have good performance. But I would nevertheless suggest using the ENet multiview features considering the diluted details of the low-res mesh.

**Comment by yang-Michael** [comment]
Ok, I will carefully consider your reply and try to run the code. Thanks again for your patience.



Minyang Xu
WUHAN UNIVERSITY&nbsp;&nbsp;
***@***.***
Number:18571459519


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "daveredrum/Scan2Cap"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月26日(星期一) 晚上7:40
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [daveredrum/Scan2Cap] Caught KeyError in DataLoader worker process 0. (#3)





 
Replacing the --use_multiview flag with --use_color is absolutely valid - Note you might want to add --use_normal to your current command, as using raw RGB values plus point normals are shown to have good performance. But I would nevertheless suggest using the ENet multiview features considering the diluted details of the low-res mesh.

**Comment by yang-Michael** [comment]
`python scripts/train.py --use_multiview --use_normal --use_topdown --use_relation --use_orientation --num_graph_steps 2 --num_locals 10 --batch_size 12 --epoch 50`
@daveredrum Hello, Dr. Chen, TODAY I changed to a Windows computer, downloaded and used all the data (including the data in the fifth part), and executed the end-to-end training command mentioned above, but the problem as shown in the following figure still appeared. It is the same as the problem I mentioned at the beginning, whether the debug mode is used or not.
陈博士好，我今天换了一台内存和显存都很大的工作站，并且使用了所有的数据，包括第五部中的可选数据。根据Readme一步步操作，执行上述所示的端到端训练命令(End-to-End training for 3D dense captioning)，并且没有使用--debug-mode参数，但是仍然报了如下图所示的同样的错误。我认为错误的原因是跟nyu40_labels.csv这个excel表有关，但是我没有找到错误的根本原因，所以又来请教一下。

![image](https://user-images.githubusercontent.com/29976881/127103202-e4fe637f-6b75-4288-97a7-55ae47475cdf.png)

**Comment by xwhjy** [comment]
> `python scripts/train.py --use_multiview --use_normal --use_topdown --use_relation --use_orientation --num_graph_steps 2 --num_locals 10 --batch_size 12 --epoch 50`
> @daveredrum Hello, Dr. Chen, TODAY I changed to a Windows computer, downloaded and used all the data (including the data in the fifth part), and executed the end-to-end training command mentioned above, but the problem as shown in the following figure still appeared. It is the same as the problem I mentioned at the beginning, whether the debug mode is used or not.
> 陈博士好，我今天换了一台内存和显存都很大的工作站，并且使用了所有的数据，包括第五部中的可选数据。根据Readme一步步操作，执行上述所示的端到端训练命令(End-to-End training for 3D dense captioning)，并且没有使用--debug-mode参数，但是仍然报了如下图所示的同样的错误。我认为错误的原因是跟nyu40_labels.csv这个excel表有关，但是我没有找到错误的根本原因，所以又来请教一下。
> 
> ![image](https://user-images.githubusercontent.com/29976881/127103202-e4fe637f-6b75-4288-97a7-55ae47475cdf.png)


Hi Michael, 

The error message says `keyError: 0` and it's indicating you are trying to fetch a value whose key value is 0 from the dictionary DC.nyu40id2class but there isn't such key value. Maybe try to look into your instance_bboxes files to see whether there is any 0 for the semantic label column [:, -2] which is not supposed to be the case since 0 means background and you couldn't have a background object right?

**Comment by yang-Michael** [comment]
> > `python scripts/train.py --use_multiview --use_normal --use_topdown --use_relation --use_orientation --num_graph_steps 2 --num_locals 10 --batch_size 12 --epoch 50`
> > @daveredrum Hello, Dr. Chen, TODAY I changed to a Windows computer, downloaded and used all the data (including the data in the fifth part), and executed the end-to-end training command mentioned above, but the problem as shown in the following figure still appeared. It is the same as the problem I mentioned at the beginning, whether the debug mode is used or not.
> > 陈博士好，我今天换了一台内存和显存都很大的工作站，并且使用了所有的数据，包括第五部中的可选数据。根据Readme一步步操作，执行上述所示的端到端训练命令(End-to-End training for 3D dense captioning)，并且没有使用--debug-mode参数，但是仍然报了如下图所示的同样的错误。我认为错误的原因是跟nyu40_labels.csv这个excel表有关，但是我没有找到错误的根本原因，所以又来请教一下。
> > ![image](https://user-images.githubusercontent.com/29976881/127103202-e4fe637f-6b75-4288-97a7-55ae47475cdf.png)
> 
> Hi Michael,
> 
> The error message says `keyError: 0` and it's indicating you are trying to fetch a value whose key value is 0 from the dictionary DC.nyu40id2class but there isn't such key value. Maybe try to look into your instance_bboxes files to see whether there is any 0 for the semantic label column [:, -2] which is not supposed to be the case since 0 means background and you couldn't have a background object right?

@xwhjy Yes, that's the reason. The fundamental problem is that I downloaded the wrong aggregate.json file from scannet.
